aws-iam-policies/domain-protect-audit.json

```diff
@@ -13,6 +13,16 @@
         "route53domains:ListDomains"
       ]
     },
+	{
+	  "Sid": "DomainProtectSecret",
+	  "Effect": "Allow",
```

Comment:
Permissions to access the secret are not required here, as these are the permissions for Domain Protect to use in every AWS account within the org, whereas the secret is only accessed within the security tooling account. 

I'd suggest a new policy is required in this directory which would then be called to add the permissions to the IAM role assumed by the Lambda function.

https://github.com/domain-protect/domain-protect/tree/main/terraform-modules/iam/templates

The resource pattern needs to specify the environment using `local.env` to avoid giving a `dev` environment access to `prd` secrets, in the case that `dev` and `prd` are both deployed to the same AWS account

Reply:
how do the templates in https://github.com/domain-protect/domain-protect/tree/main/terraform-modules/iam/templates work? Can I do conditional check with tpl file? Tbh, I dont think I used the iam module when I deployed domain-protect. Why do we have both `aws-iam-policies` and `iam/templates`?

Reply:
I think I kinda know the answer to my questions as `aws-iam-policies` are suggested permissions for the security_audit role but yeah still a bit confused.

Reply:
the purpose of the AWS IAM example policies are described at (https://github.com/domain-protect/domain-protect/blob/main/docs/aws-iam-policies.md)

Reply:
the [IAM policy templates](https://github.com/domain-protect/domain-protect/tree/main/terraform-modules/iam/templates) are used to build the IAM policies attached to IAM roles assumed by the Lambda functions. They are selected from the top level [main.tf](https://github.com/domain-protect/domain-protect/blob/main/main.tf) using the `policy` Terraform variable

Reply:
values are passed into the policy template on line 11 of [this file](https://github.com/domain-protect/domain-protect/blob/main/terraform-modules/iam/main.tf)

Reply:
yeah, I got it. Do you mind testing this once you have time? Also, what happened to all the checks?